pkg/sources/git/git.go

```diff
@@ -327,10 +327,16 @@ func (s *Source) scanDir(ctx context.Context, gitDir string, reporter sources.Ch
 		// TODO: Figure out why we skip directories ending in "git".
 		return nil
 	}
+
+	// Check if the directory exists
+	if _, err := os.Stat(gitDir); os.IsNotExist(err) {
+		return fmt.Errorf("directory does not exist: %s", gitDir)
+	}
```

Comment:
I believe this change is the only one that's needed. With just it applied, this is the output I get:
```
trufflehog (remotes/origin/HEAD *) % ./trufflehog git file:///oops
🐷🔑🐷  TruffleHog. Unearth your secrets. 🐷🔑🐷

2024-10-16T16:14:08-04:00	info-0	trufflehog	running source	{"source_manager_worker_id": "hN9fy", "with_units": true}
2024-10-16T16:14:08-04:00	error	trufflehog	error running scan	{"error": "engine failed to finish execution: fatal: directory does not exist: /oops"}
trufflehog (remotes/origin/HEAD *) % echo $?
1
```
Are you seeing something different?